### DISCUSSION TITLE
How to get it to work for standard account? (Intune)

### DISCUSSION THREAD

**Discussion by fmiskic** [Romanitho/Winget-Install#59] [Q&A]
Hello,

Maybe I am missing the purpose. Is this supposed to work for non-admin accounts? I've tested with an admin account and it seems to deploy fine. As soon as it's a standard user it fails.

I mean I understand why this is the case since the PATH for app installer is default to the standard account of the machine. Once you elevate to an admin account you are running it as the elevated account. The problem with even adding the standard account winget to your path is that it's missing the database.

Any help would be appreciated if I am missing something.

Cheers

**Comment by KnifMelti** [comment]
Hi!
An installation of a **System** app via `Winget-Install` must in **Intune** be deployed to Users/Clients as **Run in System context**.
In the **Company Portal** it can be either **Available** or **Required**.
I.e. you must make an application in Intune and assign it...

**Reply by Romanitho** [reply]
This project was born to install apps as system or elevated user account. If you want to install an app with intune to the user with user rights, I guess you can simply call `winget install appdID` (but not tried yet ^^)

**Reply by KnifMelti** [reply]
...but then the app itself must support an install in user context...

**Reply by fmiskic** [reply]
> Hi! An installation of a **System** app via `Winget-Install` must in **Intune** be deployed to Users/Clients as **Run in System context**. In the **Company Portal** it can be either **Available** or **Required**. I.e. you must make an application in Intune and assign it...

I did not quite follow. So to install apps that are System apps. On Intune it needs to user and not system even though it is a system app?

**Reply by KnifMelti** [reply]
If you add a win32 app in Intune you set if it should install in user or system context.
Only apps supporting installing in user context can be run under user context.
An app installing in system context can be deployed/assigned to users or clients, but it will be installed for System.

**Reply by Romanitho** [reply]
In any case, i would suggest to install for system, except the case the app is not system installable

**Reply by fmiskic** [reply]
> If you add a win32 app in Intune you set if it should install in user or system context. Only apps supporting installing in user context can be run under user context. An app installing in system context can be deployed/assigned to users or clients, but it will be installed for System.

Yes so this is what I mean. For example trying to install Firefox or NordVPN under system context just fails when installing from the company portal. This is on a computer with a standard user. When it was tried with a computer that had an admin account requesting the installation the same script worked. 

So is the intention of the script to be able to install from a standard non-admin account? It seems that it's not working.

**Reply by KnifMelti** [reply]
Very strange...
The intention (how it works) is that of installing apps under System context for all users, otherwise there's no point in it.
I'm installing everything (haven't tried FF/NordVPN yet, though 😉) in our environment...
...could it be the PS x86/x64 @Romanitho

**Reply by Romanitho** [reply]
I tried to deploy Firefox under system with Winget on a sandbox, and it works. Firefox get installed and detected. So, supported.
Could you share with us the config you have ?

**Reply by fmiskic** [reply]
> I tried to deploy Firefox under system with Winget on a sandbox, and it works. Firefox get installed and detected. So, supported. Could you share with us the config you have ?

Thanks for testing this. Again though is this working with a non-admin account? It worked for me as well when the user had admin.

![Screenshot 2022-10-17 at 13 55 04](https://user-images.githubusercontent.com/48210659/196170864-deb62aab-a941-407b-a90e-e4ee0e19413e.png)

**Comment by KnifMelti** [comment]
Intune Deployment with `%systemroot%\sysnative\WindowsPowerShell`:
https://github.com/Romanitho/Winget-AutoUpdate/discussions/88
https://www.anoopcnair.com/intune-win32-app-deploy-system32-vs-syswow64/

**Reply by fmiskic** [reply]
This is what was in place:
`"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -ExecutionPolicy Bypass -File winget-install.ps1 -AppIDs Mozilla.Firefox`

**Reply by KnifMelti** [reply]
FF is in a WAU thread too:
https://github.com/Romanitho/Winget-AutoUpdate/issues/158

Have you tried the ESR version?

**Reply by Romanitho** [reply]
you need to be sure it's configured for system install

**Reply by fmiskic** [reply]
> FF is in a WAU thread too: [Romanitho/Winget-AutoUpdate#158](https://github.com/Romanitho/Winget-AutoUpdate/issues/158)
> 
> Have you tried the ESR version?

I will give this a try and get back to you.

**Reply by KnifMelti** [reply]
But @Romanitho used the ordinary FF, I suppose...

**Reply by KnifMelti** [reply]
Any luck @fmiskic ?

**Reply by fmiskic** [reply]
@KnifMelti Unfortunately I have not. Once I do I will report back.

**Reply by fmiskic** [reply]
> Any luck @fmiskic ?

It seems the ESR also fails. Same script just Mozilla.Firefox.ESR

**Comment by trandall50** [comment]
I'm in the same boat. Seems to be the only way to install teams standalone (winget+intune) but looks like it will be all for not if it won't work for a standard user. I was so close with this script but it appears it will be back to the drawing board. I can't believe how difficult it is deploy teams to standard users without installing 365 suite.. I'm blown away.

**Reply by KnifMelti** [reply]
https://learn.microsoft.com/en-us/microsoftteams/msi-deployment

**Reply by trandall50** [reply]
Thanks for the link. I was hoping to achieve with winget though

**Reply by KnifMelti** [reply]
Ok, and that's a problem?
I'm not fond of Intune, using SCCM for application deployment even if we are hybrid joined.
We're using Office, but I'll create a WinGet app for Teams myself and test.
We only have standard users in our environment.

**Reply by KnifMelti** [reply]
This looks like the one:
`winget show Microsoft.Teams.ContinuousDeployment`
https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/Teams/ContinuousDeployment/1.5.00.31365

**Reply by KnifMelti** [reply]
Installed manually via WinGet as a standard user:
![image](https://github.com/Romanitho/Winget-Install/assets/102996177/50afe535-2fa1-4189-8b4e-cf3a0b9e72c6)

**Reply by Romanitho** [reply]
User context should work yes.

**Comment by KnifMelti** [comment]
On a Swedish system...

![image](https://github.com/Romanitho/Winget-Install/assets/102996177/6126816e-6d1d-41a3-8552-1fcbcad7362d)

![image](https://github.com/Romanitho/Winget-Install/assets/102996177/ed17b3d9-201b-4594-b3e2-02474025c356)

![image](https://github.com/Romanitho/Winget-Install/assets/102996177/5f1c2a69-039a-41a8-a44e-90b04766eac3)

![image](https://github.com/Romanitho/Winget-Install/assets/102996177/a188ed27-6cdf-45eb-b28e-0a2adbdc76df)

![image](https://github.com/Romanitho/Winget-Install/assets/102996177/cd845bd0-f234-4b83-a7f7-02c9616a5d72)

**Reply by trandall50** [reply]
This confirms it then!? winget-install.ps1 works just fine for user published packages (MS teams) and standard users won't run into issues executing from company portal it appears! This is great! The initial post had me worried, but the thorough reply has me confident now. 

THANK YOU for your good work!!

**Reply by KnifMelti** [reply]
NB: in the detection script you must use `Microsoft.Teams` (not `Microsoft.Teams.ContinuousDeployment`) - took me a while to crack that one

**Reply by trandall50** [reply]
yeah good catch. makes sense

**Reply by KnifMelti** [reply]
...or, **NOT!** 😂 (since the WinGet ID is `Microsoft.Teams.ContinuousDeployment`)

**Reply by KnifMelti** [reply]
When `Install behavior = User` in **Intune** `winget-install.ps1` shows the PowerShell execution in a user session window but if the application is a System installation and `Install behavior = System` you don't see anything.

**Reply by Romanitho** [reply]
You can use hidden powershell (small PS window flashes).
I don't get why the id is not the same on direction?

**Reply by KnifMelti** [reply]
Tried -NoNewWindow, but that stopped it from working

**Reply by KnifMelti** [reply]
Ah, -WindowStyle hidden

**Reply by Romanitho** [reply]
Yep.
You won't be able to use -NoNewWindow as process switches between 32 and 64.

Last shot, use the invisible.vbs from WAU. Same command as user context scheduled task

**Reply by KnifMelti** [reply]
> I don't get why the id is not the same on direction?

Microsoft 😍
